packages/kbn-apm-synthtrace/src/cli/utils/index_schedule.ts

```diff
@@ -0,0 +1,308 @@
+            momentInstance: moment,
+          })!
+          .valueOf();
+    const endTs = isNumber(schedule.end)
+      ? schedule.end
+      : isString(schedule.end)
+      ? datemath
+          .parse(schedule.end, {
+            roundUp: false,
+            momentInstance: moment,
+          })!
+          .valueOf()
+      : false;
+    return { ...schedule, start: startTs, end: endTs };
+  };
+
+interface Point {
+  x: number;
+  y: number;
+}
+
+function createExponentialFunction(start: Point, end: Point) {
+  const totalPoints = end.x - start.x;
+  const ratio = end.y / start.y;
+  const exponent = Math.log(ratio) / (totalPoints - 1);
+  return (timestamp: Moment) => {
+    const x = timestamp.valueOf() - start.x;
+    return Math.round(start.y * Math.exp(exponent * x));
+  };
+}
+
+function createSineFunction(start: Point, end: Point, period = 60) {
+  const midline = start.y;
+  const amplitude = end.y;
+  return (timestamp: Moment) => {
+    const x = (timestamp.valueOf() - start.x) / 1000;
+    const y = midline + amplitude * Math.sin(((2 * Math.PI) / period) * x);
+    if (y < 0) {
+      return 0;
+    }
+    return y;
+  };
+}
+
+function createLinearFunction(start: Point, end: Point) {
+  const slope = (end.y - start.y) / (end.x - start.x);
+  const intercept = start.y - slope * start.x;
+  return (timestamp: Moment) => {
+    return slope * timestamp.valueOf() + intercept;
+  };
+}
+
+function createEventsPerCycleFn(
+  schedule: ParsedSchedule,
+  eventsPerCycle: EventsPerCycle
+): (timestamp: Moment) => number {
+  if (EventsPerCycleTransitionDefRT.is(eventsPerCycle) && isNumber(schedule.end)) {
+    const startPoint = { x: schedule.start, y: eventsPerCycle.start };
+    const endPoint = { x: schedule.end, y: eventsPerCycle.end };
+    if (eventsPerCycle.method === 'exp') {
+      return createExponentialFunction(startPoint, endPoint);
+    }
+    if (eventsPerCycle.method === 'sine') {
+      return createSineFunction(startPoint, endPoint, 60); // TODO add  eventsPerCycle.options?.period
+    }
+    return createLinearFunction(startPoint, endPoint);
+  } else if (EventsPerCycleTransitionDefRT.is(eventsPerCycle) && schedule.end === false) {
+    // eslint-disable-next-line no-console
+    console.log('EventsPerCycle must be a number if the end value of schedule is false.');
+  }
+
+  return (_timestamp: Moment) =>
+    typeof eventsPerCycle === 'number' ? eventsPerCycle : eventsPerCycle.end;
+}
+
+export async function wait(delay: number, logger: Logger) {
+  logger.info(`Waiting ${delay}ms`);
+  await new Promise((resolve) => {
+    setTimeout(resolve, delay);
+  });
+}
+
+export async function createEvents(
+  config: Config,
+  schedule: ParsedSchedule,
+  end: Moment | false,
+  currentTimestamp: Moment,
+  continueIndexing = false,
+  logger: Logger,
+  stream: PassThrough,
+  apmEsClient: ApmSynthtraceEsClient
+): Promise<void> {
+  const scenarioFile = `../../config/scenarios/${schedule.scenario}`;
```

Comment:
I created a new scenarios folder under the new config folder instead of using the existing `src/scenarios` folder. What do you think?

Reply:
@mgiota Looking at the scenario file, they look like quite generic scenarios or am i missing something ?
If they are generic scenarios which simply run with different configs, we can have them in the same scenario folder. This would mean any scenarios could be run using this config.

Reply:
@achyutjhunjhunwala Yep they are generic scenarios and ideally I would like them in the existing scenarios folder. I just put them in a new folder as an initial implementation. And you are right, any scenarios could be using this config. Ok cool, I will update my code to read the scenarios from the existing scenarios folder.

@almudenasanz Do you have a specific scenario and the corresponding configuration you want me to include? For example I was thinking to delete the good and bad configurations I added in this PR and replace them with something more real:

```
schedule:
  - scenario: "high_throughput"
    start: "now-50m"
    end: "now-45m",
  - scenario: "low_throughput"
    start: "now-45m"
    end: "now-40m"
  - scenario: "high_throughput"
    start: "now-40m"
    end: false
```

Reply:
@mgiota ideally, the scenarios would cover the APM rules: 
- Latency
- Error count 
- Failed transaction rate
In the original script we had for the ON week, it covered the Error count and Failed Transaction Rate, we were missing an extra scenario for latency